File: cap-1/python/pesquisa-binaria.py
def pesquisa_binaria(lista, item):
    baixo = 0
    alto = len(lista) - 1

    while baixo <= alto:
        meio = (baixo + alto) // 2
        chute = lista[meio]
        if chute == item:
            return meio
        if chute > item:
            alto = meio - 1
        else:
            baixo = meio + 1
    return None


minha_lista = list(range(1, 1000001))

print(pesquisa_binaria(minha_lista, 756000))
print(pesquisa_binaria(minha_lista, -1))

File: cap-1/javascript/pesquisa-binaria.js
function PesquisaBinaria(lista, item) {
    let baixo = 0
    let alto = lista.length - 1

    while (baixo <= alto) {
        let meio = Math.floor((baixo + alto) / 2)
        let chute = lista[meio]
        if(chute === item) {
            return meio
        }
        if(chute > item) {
            alto = meio - 1
        } else {
            baixo = meio + 1
        }
    }
    return undefined
}

minha_lista = Array.from({length: 1000000}, (_, i)=> i+ 1);
console.log(PesquisaBinaria(minha_lista, 756000))
console.log(PesquisaBinaria(minha_lista, -1))

